app/controllers/concerns/sign_in/authentication.rb

```diff
@@ -34,6 +34,14 @@ def load_user(skip_expiration_check: false)
       nil
     end
 
+    def access_token_authenticate(skip_expiration_check: false)
```

Comment:
You'll want to `handle_authenticate_error` for the case when `revoke_all_sessions` calls this, which makes me wonder if the case when `revoke_all_sessions` is calling this is different enough from when `logout` is calling it that they should be different functions.

Maybe `access_token_authenticate` as you have here (although no need for the `skip_expiration_check`, and also needs to throw relevant errors when `StandardError` is hit) which `revoke_all_sessions` will call, and then something like `access_token_optional_authenticate`, which `logout` will call